File: src/main/java/com/terraforming/ares/model/request/BuildProjectRequest.java
package com.terraforming.ares.model.request;

import com.terraforming.ares.model.payments.Payment;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

/**
 * Created by oleksii.nikitin
 * Creation date 12.05.2022
 */
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
public class BuildProjectRequest {
    private String playerUuid;
    private Integer cardId;
    private List<Payment> payments;
    private Map<Integer, List<Integer>> inputParams;
}

File: frontend/src/app/data/BuildProjectRequest.ts
import {Payment} from './Payment';

export class BuildProjectRequest {
  constructor(playerUuid: string, cardId: number, payments: Payment[], inputParams?: Map<number, number[]>) {
    this.playerUuid = playerUuid;
    this.cardId = cardId;
    this.payments = payments;
    this.inputParams = inputParams;
  }

  playerUuid: string;
  cardId: number;
  payments: Payment[];
  inputParams: Map<number, number[]>;

}
